ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/router_bgp.py

```diff
@@ -94,7 +90,13 @@ def router_bgp(self) -> dict | None:
                 if self._filter_peer_as is True:
                     neighbor["route_map_out"] = f"RM-BGP-AS{link['peer_bgp_as']}-OUT"
 
-                neighbors[link["peer_ip_address"]] = neighbor
+                if link["peer_ip_address"] in ip_addresses:
+                    for idx, neighbor in enumerate(neighbors):
+                        if neighbor["ip_address"] == link["peer_ip_address"]:
+                            neighbors[idx] = neighbor
+                else:
+                    ip_addresses.append(link["peer_ip_address"])
+                    neighbors.append(neighbor)
```

Comment:
@ClausHolbechArista - it seems we are overwriting the neighbor config (as we used to do) if we have found it already - should we put in the duplicate error as we did for port-channels on network ports?

@Shivani-chourasiya - the code could be simplified with get_item I think

Reply:
Yes good idea to catch the issue and raise instead of just overwriting.

Reply:
okay

Reply:
Updated the code, but getting duplication error on some moleucle run